packages/aws-cdk/test/notices.test.ts

```diff
@@ -0,0 +1,291 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as nock from 'nock';
+import {
+  CachedDataSource,
+  filterNotices,
+  formatNotices,
+  generateMessage,
+  Notice,
+  WebsiteNoticeDataSource,
+} from '../lib/notices';
+
+const BASIC_NOTICE = {
+  title: 'Toggling off auto_delete_objects for Bucket empties the bucket',
+  issueNumber: 16603,
+  overview: 'If a stack is deployed with an S3 bucket with auto_delete_objects=True, and then re-deployed with auto_delete_objects=False, all the objects in the bucket will be deleted.',
+  components: [{
+    name: 'cli',
+    version: '<=1.126.0',
+  }],
+  schemaVersion: '1',
+};
+
+const MULTIPLE_AFFECTED_VERSIONS_NOTICE = {
+  title: 'Error when building EKS cluster with monocdk import',
+  issueNumber: 17061,
+  overview: 'When using monocdk/aws-eks to build a stack containing an EKS cluster, error is thrown about missing lambda-layer-node-proxy-agent/layer/package.json.',
+  components: [{
+    name: 'cli',
+    version: '<1.130.0 >=1.126.0',
+  }],
+  schemaVersion: '1',
+};
+
+const FRAMEWORK_2_1_0_AFFECTED_NOTICE = {
+  title: 'Regression on module foobar',
+  issueNumber: 1234,
+  overview: 'Some bug description',
+  components: [{
+    name: 'framework',
+    version: '<= 2.1.0',
+  }],
+  schemaVersion: '1',
+};
+
+describe('cli notices', () => {
+  describe(formatNotices, () => {
+    test('correct format', () => {
+      expect(formatNotices([BASIC_NOTICE])).toEqual([
+        [
+          `${BASIC_NOTICE.issueNumber}\t${BASIC_NOTICE.title}`,
```

Comment:
I know I always say "no shapsnots" but I think testing formatting functions is the one place where snapshot tests make sense!